pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -516,16 +520,36 @@ impl ImageLayerInner {
                 let key = Key::from_slice(&raw_key[..KEY_SIZE]);
                 assert!(key >= range.start);
 
+                let drop = !filter(&key);
```

Comment:
I think the code below is correct, but I find it hard to reason about.
You could do the following instead:
```
let flag = {
  if !filter(&key) {
    BlobFlag::Ignore
  } else {
    BlobFlag::None
  }
}
```

Pass this flag into `planner.handle` and don't modify the rest of the code.
This offloads the work to the planer and remove s the need for separate tests.

Reply:
ah, a very nice suggestion.  c54ca636995b18fbe04425f0fff35cb030e14827